### PR TITLE
Major version: v3

### PR DESCRIPTION
**Pull request by patinthehat**
This PR updates the package to a new major version, `v3.0.0`.

### Major changes:
- Dropped support for PHP versions < 8.2.
- Many breaking API changes (see below).
- Added support for saving multiple pages to images using `selectPages()`.
- Native Backed Enum implementations to enforce allowed values in several places.
- Dropped support for reading remote PDF files for security reasons.
- Added property, argument and return types.
- Removed Imagick instance creation in constructor, replaced calls to `readImage()` with `pingImage()` _(test suite runs ~33% faster)_.

### Breaking API Changes
There are breaking API changes from v2 to v3. I feel that these changes streamline the developer experience, enhance understanding of the functionality of the methods, and in some cases reduce cognitive overload by simplifying method names.  See the modified `README.md` for a full list of the new API methods and method names.

### Additional Changes
- `php-cs-fixer` configuration and workflow was dropped in favor of Laravel Pint.
- Readme updated to include all available methods, fix a few grammatical errors, etc.
- Fixed several minor, previously unknown bugs.
- Test coverage increased from ~65% to ~93%.
- Pest upgraded to v2, migrated PHPUnit configuration to v10.
- Unit tests were reorganized/restructured to be more manageable, and Pest configuration files were added.

### CODE REVIEW

File: src/DTOs/PdfPage.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        $this->prefix = $prefix;
+        $this->path = $path;
+    }
+
+    public static function make(int $pageNumber, OutputFormat $format, string $prefix, string $path): self
+    {
+        return new self($pageNumber, $format, $prefix, $path);
+    }
+
+    public function getFilename(): string
```
Comment (Nielsvanpach) [inline src/DTOs/PdfPage.php:30]: What about dropping the get prefix, and just using fileName() ?

**Review by Nielsvanpach — COMMENTED** [review]
Looks good!

I just had a quick scan through the code, as I'm not familiar with this package.

File: src/DTOs/PdfPage.php (hunk 1 above)
Comment (patinthehat) [reply src/DTOs/PdfPage.php:30]: That's a great suggestion and in line with the other method names, thanks! :+1:

File: src/DTOs/PdfPage.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public readonly string $path;
+
+    public function __construct(int $pageNumber, OutputFormat $format, string $prefix, string $path)
+    {
+        $this->number = $pageNumber;
+        $this->format = $format;
+        $this->prefix = $prefix;
+        $this->path = $path;
+    }
```
[the remark this answers is not in the document]
Comment (patinthehat) [reply src/DTOs/PdfPage.php:23]: That's a better idea, thanks!

**Comment by patinthehat** [comment]
Hey @freekmurze or @AlexVanderbist, please review this new major version PR if time permits. Thanks! :+1:

File: src/Enums/LayerMethod.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Spatie\PdfToImage\Enums;
+
+use Imagick;
+
+enum LayerMethod: int
```
Comment (freekmurze) [inline src/Enums/LayerMethod.php:7]: Very nice!

File: src/Exceptions/InvalidLayerMethod.php [hunk 4]
```diff
@@ -6,4 +6,8 @@
 
 class InvalidLayerMethod extends Exception
 {
+    public static function for(int $value)
```
Comment (freekmurze) [inline src/Exceptions/InvalidLayerMethod.php:9]: Let's add a return type hint here as well

File: src/Exceptions/InvalidSize.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Spatie\PdfToImage\Exceptions;
+
+class InvalidSize extends \Exception
```
Comment (freekmurze) [inline src/Exceptions/InvalidSize.php:5]: Nitpick: let's import `Exception` so we can lose that `\`

File: src/Pdf.php [hunk 6]
```diff
@@ -161,75 +211,106 @@ public function getImageData(string $pathToImage): Imagick
...
-
         if (! $this->isValidOutputFormat($outputFormat)) {
-            $outputFormat = 'jpg';
+            $outputFormat = OutputFormat::Jpg;
         }
 
         return $outputFormat;
     }
+
+    protected function validatePageNumbers(int ...$pageNumbers)
```
Comment (freekmurze) [inline src/Pdf.php:306]: Let's add a return type here as well.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Very nice work! Thank you for this

File: src/DTOs/PdfPage.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Spatie\PdfToImage\DTOs;
+
+use Spatie\PdfToImage\Enums\OutputFormat;
+
+class PdfPage
+{
+    public function __construct(
```
Comment (Nielsvanpach) [inline src/DTOs/PdfPage.php:9]: A protected constructor for DTOs or VOs makes it easier to refactor or extend in the future.

```suggestion
    protected function __construct(
```

File: src/Exceptions/InvalidLayerMethod.php (hunk 4 above)
Comment (Nielsvanpach) [inline src/Exceptions/InvalidLayerMethod.php:9]:
```suggestion
    public static function for(int $value): static
```

File: src/DTOs/PdfPage.php (hunk 7 above)
Comment (patinthehat) [reply src/DTOs/PdfPage.php:9]: @Nielsvanpach Could you elaborate on this and/or is it Spatie practice to make DTO constructors protected? I'm definitely leaning towards agreement, but if you could expand a bit I'd appreciate it :+1:

File: src/Exceptions/InvalidSize.php (hunk 5 above)
Comment (patinthehat) [reply src/Exceptions/InvalidSize.php]: Nice catch! :+1:

**Pull request merged**